packages/core/src/client.ts

```diff
@@ -2627,6 +2627,52 @@ export class MedplumClient extends EventTarget {
     this.basicAuth = encodeBase64(clientId + ':' + clientSecret);
   }
 
+  /**
+   * Request oauth token with client credentials
+   *
+   * @category Authentication
+   * @param clientId The client ID.
+   * @param clientSecret The client secret.
+   */
+  async requestAccessToken(clientId: string, clientSecret: string): Promise<void> {
+    const formBody = new URLSearchParams();
+    formBody.set('grant_type', OAuthGrantType.ClientCredentials);
+    formBody.set('client_id', clientId);
+    formBody.set('client_secret', clientSecret);
+
+    const response = await this.fetch(this.tokenUrl, {
+      method: 'POST',
+      headers: { Authorization: `Basic ${encodeBase64(clientId + ':' + clientSecret)}` },
+      body: formBody,
+      credentials: 'include',
+    });
+
+    if (!response.ok) {
+      throw new Error('Failed to fetch tokens');
+    }
+    const tokens = await response.json();
+    const accessToken = tokens.access_token;
+
+    // Verify token has not expired
+    const tokenPayload = parseJWTPayload(accessToken);
+
+    if (Date.now() >= (tokenPayload.exp as number) * 1000) {
+      throw new Error('Token expired');
+    }
+
+    // Verify app_client_id
+    // external tokenPayload
+    if (tokenPayload.cid) {
+      if (tokenPayload.cid !== clientId) {
+        throw new Error('Token was not issued for this audience');
+      }
+    } else if (this.clientId && tokenPayload.client_id !== this.clientId) {
+      throw new Error('Token was not issued for this audience');
+    }
```

Comment:
could be refactored to `isTokenIssuedForAudience(tokenPayload, clientId)` so that it can be reused with [verifyTokens](https://github.com/medplum/medplum/blob/b848de434bca1de6a96cc5cf5de545acd4f13a31/packages/core/src/client.ts#L2678)